File: CloudflareDatabase.js
const sqlite3 = require('sqlite3').verbose();
const http = require('http');

const db = new sqlite3.Database('cloudflare.db');

const requestListener = (req, res) => {
    try {
        const url = new URL(req.url, `http://${req.headers.host}`);

        if (url.pathname.startsWith("/setvar")) {
            const parts = url.pathname.split('/').filter(p => p);
            if (parts.length < 3) {
                res.writeHead(400);
                res.end("Error: Key or Value missing");
                return;
            }
            const [ , key, value ] = parts;
            db.run("INSERT OR REPLACE INTO variables (key, value) VALUES (?, ?)", [key, value], (err) => {
                if (err) {
                    res.writeHead(500);
                    res.end(`Error: ${err.message}`);
                } else {
                    res.writeHead(200);
                    res.end(`Variable '${key}' set to '${value}'.`);
                }
            });
        } else if (url.pathname.startsWith("/getvar")) {
            const parts = url.pathname.split('/').filter(p => p);
            if (parts.length < 2) {
                res.writeHead(400);
                res.end("Error: Key missing");
                return;
            }
            const key = parts[1];
            db.get("SELECT value FROM variables WHERE key = ?", [key], (err, row) => {
                if (err) {
                    res.writeHead(500);
                    res.end(`Error: ${err.message}`);
                } else if (row) {
                    res.writeHead(200);
                    res.end(row.value);
                } else {
                    res.writeHead(404);
                    res.end("Variable not found");
                }
            });
        } else {
            res.writeHead(200);
            res.end("Hello from Cloudflare D1!");
        }
    } catch (error) {
        res.writeHead(500);
        res.end(`Error: ${error.message}`);
    }
};

const server = http.createServer(requestListener);
server.listen(3000, () => {
    console.log("Server running at http://localhost:3000/");
});

File: CloudflareDatabase.cs
using System;
using System.Net;
using System.Threading.Tasks;
using Microsoft.Data.Sqlite;

public class CloudflareDatabase
{
    public async Task HandleRequest(HttpListenerRequest request, HttpListenerResponse response)
    {
        try
        {
            var url = new Uri(request.Url.ToString());

            // POST: Set variable
            if (url.AbsolutePath.StartsWith("/setvar"))
            {
                var parts = url.AbsolutePath.Split('/').Where(p => !string.IsNullOrEmpty(p)).ToArray();
                if (parts.Length < 3)
                {
                    response.StatusCode = 400;
                    await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("Error: Key or Value missing"));
                    return;
                }

                string key = parts[1];
                string value = parts[2];

                using (var connection = new SqliteConnection("Data Source=cloudflare.db"))
                {
                    connection.Open();
                    var command = connection.CreateCommand();
                    command.CommandText = "INSERT OR REPLACE INTO variables (key, value) VALUES (@key, @value)";
                    command.Parameters.AddWithValue("@key", key);
                    command.Parameters.AddWithValue("@value", value);
                    command.ExecuteNonQuery();
                }

                response.StatusCode = 200;
                await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes($"Variable '{key}' set to '{value}'."));
            }
            // GET: Get variable
            else if (url.AbsolutePath.StartsWith("/getvar"))
            {
                var parts = url.AbsolutePath.Split('/').Where(p => !string.IsNullOrEmpty(p)).ToArray();
                if (parts.Length < 2)
                {
                    response.StatusCode = 400;
                    await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("Error: Key missing"));
                    return;
                }

                string key = parts[1];
                string value = null;

                using (var connection = new SqliteConnection("Data Source=cloudflare.db"))
                {
                    connection.Open();
                    var command = connection.CreateCommand();
                    command.CommandText = "SELECT value FROM variables WHERE key = @key";
                    command.Parameters.AddWithValue("@key", key);
                    using (var reader = command.ExecuteReader())
                    {
                        if (reader.Read())
                        {
                            value = reader.GetString(0);
                        }
                    }
                }

                if (value != null)
                {
                    response.StatusCode = 200;
                    await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes(value));
                }
                else
                {
                    response.StatusCode = 404;
                    await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("Variable not found"));
                }
            }
            else
            {
                response.StatusCode = 200;
                await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes("Hello from Cloudflare D1!"));
            }
        }
        catch (Exception e)
        {
            response.StatusCode = 500;
            await response.OutputStream.WriteAsync(System.Text.Encoding.UTF8.GetBytes($"Error: {e.Message}"));
            
        }
    }
}
